### PR TITLE
make stuff compact

### PR DESCRIPTION
**Pull request by KristofferC**
Changes all the nested vectors to pack their data compactly. Easier to read and better performance imo.

@fredrikekre

### CODE REVIEW

**Comment by fredrikekre** [comment]
Out of curiosity; why the change from `Vector` to `AbstractVector` on all the places?

**Comment by KristofferC** [comment]
Because I pass in a subarray when creating fevalues which is not a vector.

File: src/commons_abstract_fevalues.jl [hunk 1]
```diff
@@ -186,21 +177,19 @@ where ``\\mathbf{u}_i`` are the nodal values of ``\\mathbf{u}``.
...
 
 @inline function function_gradient{dim, T}(fe_v::AbstractFEValues{dim}, q_point::Int, u::Vector{Vec{dim, T}})
     n_base_funcs = getnbasefunctions(getfunctionspace(fe_v))
     @assert length(u) == n_base_funcs
-    dN = shape_gradient(fe_v, q_point)
     grad = zero(Tensor{2, dim, T})
-    for i in 1:n_base_funcs
-        grad += u[i] ⊗ dN[i]
+    @inbounds for i in 1:n_base_funcs
+        grad += u[i] ⊗ shape_gradient(fe_v, q_point, i)
```
Comment (fredrikekre) [inline src/commons_abstract_fevalues.jl]: This method won't work I think. Atleast the \otimes should be a regular *
the input `u` should also now be a regular vector of values, not a vector of `Vec`'s.
We can probably just combined the two `function_gradient` methods now.

File: src/commons_abstract_fevalues.jl [hunk 2]
```diff
@@ -150,9 +142,8 @@ end
     n_base_funcs = getnbasefunctions(getfunctionspace(fe_v))
     @assert length(u) == n_base_funcs
     vec = zero(Vec{dim, T})
-    N = shape_value(fe_v, q_point)
     @inbounds for i in 1:n_base_funcs
-        vec += N[i] * u[i]
+        vec += shape_value(fe_v, q_point, i) * u[i]
```
Comment (fredrikekre) [inline src/commons_abstract_fevalues.jl]: same here, the `u` needs to be a regular vector.

File: src/commons_abstract_fevalues.jl (hunk 1 above)
Comment (KristofferC) [inline src/commons_abstract_fevalues.jl]: Are these comments regarding #88?

Comment (KristofferC) [inline src/commons_abstract_fevalues.jl]: FWIW, I still feel it makes sense to send in a vector of `Vec`s.

Comment (fredrikekre) [inline src/commons_abstract_fevalues.jl]: Well, we still have to rewrite the code.

Comment (KristofferC) [inline src/commons_abstract_fevalues.jl]: yes

**Pull request merged**